### PR TITLE
fix: resolve secp256k1 to latest version

### PR DESCRIPTION
**Pull request by NEllusion**
## **Description**
This pull request resolves the secp256k1 library to the latest version. Currently we use a variety of versions under 4.0.x, this would unify them under 4.0.4.

This is the state of the library before this resolution - as you can see, only patch changes will apply. 

 

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by legobeat — DISMISSED** [review]

**Comment by legobeat** [comment]
@metamaskbot update-policies

**Review by legobeat — APPROVED** [review]

**Comment by legobeat** [comment]
@NicholasEllul related:
- https://github.com/MetaMask/metamask-extension/pull/27973

File: package.json [hunk 1]
```diff
@@ -264,7 +264,8 @@
     "@metamask/network-controller@npm:^17.0.0": "patch:@metamask/network-controller@npm%3A21.0.0#~/.yarn/patches/@metamask-network-controller-npm-21.0.0-559aa8e395.patch",
     "@metamask/network-controller@npm:^19.0.0": "patch:@metamask/network-controller@npm%3A21.0.0#~/.yarn/patches/@metamask-network-controller-npm-21.0.0-559aa8e395.patch",
     "@metamask/network-controller@npm:^20.0.0": "patch:@metamask/network-controller@npm%3A21.0.0#~/.yarn/patches/@metamask-network-controller-npm-21.0.0-559aa8e395.patch",
-    "path-to-regexp": "1.9.0"
+    "path-to-regexp": "1.9.0",
+    "secp256k1": "4.0.4"
```
Comment (legobeat) [inline package.json:268]: Hm, this also tries to pin the 3.x versions, which should go to `3.8.1`?

**Review by legobeat — CHANGES_REQUESTED** [review]

**Review by danjm — APPROVED** [review]
LGTM (unless the answer to the question from Nic to lego changes my understanding of what exactly needs to be pinned here)

**Comment by legobeat** [comment]
@danjm Maybe we can consider superseding this by #11932?

**Review by FrederikBolding — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request closed without merging**